doc/ipythonconsole.rst

```diff
@@ -2,91 +2,159 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   The port number is the one on which the SSH daemon (``sshd``) is running, typically 22 unless you or your administrator has configured it otherwise.
+To connect to a kernel on a remote machine,
 
-.. image:: images/console/console-dialog-connect.png
-   :alt: Connect to kernel dialog, requesting path and connection details
+#. Launch a Spyder kernel on the remote host if one is not already running, with ``python -m spyder_kernels.console``.
 
-For more technical details about connecting to remote IPython kernels, see the `Connecting to a remote kernel`_ page in the IPython Cookbook.
-Just remember to enter the appropriate details into Spyder's :guilabel:`Connect to an existing kernel` dialog instead of launching a new frontend on the client with ``--existing``.
+   .. image:: images/console/console-connect-remote-step1.gif
+      :alt: Staring a Spyder kernel on a remote machine
 
-.. _Connecting to a remote kernel: https://github.com/ipython/ipython/wiki/Cookbook:-Connecting-to-a-remote-kernel-via-ssh
+#. Copy the kernel's connection file (:file:`{jupyter/runtime/dir/path}/kernel-{pid}.json`) to the machine you're running Spyder on.
 
+   You can get :file:`{jupyter/runtime/dir/path}` by executing ``jupyter --runtime-dir`` in the same Python environment as the kernel.
+   Usually, the connection file you are looking for will be one of the newest in this directory, corresponding to the time you started the external kernel.
 
-.. _console-features:
+   .. image:: images/console/console-connect-remote-step2.gif
+      :alt: Using SCP to copy the connection file to the local machine
 
-==================
-Supported features
-==================
+#. Click :guilabel:`Connect to an existing kernel` from the :guilabel:`Consoles` menu, and browse for or enter the path to the connection file from the previous step.
 
-Any :guilabel:`IPython Console` in Spyder, internally or externally created, supports additional features including:
+   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to :file:`{jupyter/runtime/dir/path}/kernal-{id}.json` on your local machine, if you've copied the connection file there.
 
-.. image:: images/console/console-completion.png
-   :alt: Spyder IPython Console, with a popup list of code completion guesses
+   .. image:: images/console/console-connect-remote-step3.gif
+      :alt: Opening the connect to kernel dialog and browsing for the path
 
-* Automatic code completion
-* Real-time function calltips
-* Debugging toolbar integration for launching the debugger and controlling execution flow
+#. Check the :guilabel:`This is a remote kernel (via SSH)` box and type the full hostname you're connecting to (in the form :file:`{username}@{hostname}:{port-number}`).
+   Then, enter *either* :file:`{username}`'s password on the remote machine, or your user SSH keyfile (typically a  :file:`.perm` file); only one is needed to connect.
```

Comment:
```suggestion
#. Check the :guilabel:`This is a remote kernel (via SSH)` box and enter the :guilabel:`Hostname` or IP address, username and port to connect to on the remote machine.
   Then, enter *either* :file:`{username}`'s password on the remote machine, or browse to an SSH keyfile (typically in the :file:`.ssh` directory in your home folder on the local machine, often called :file:`id_rsa` or similar) registered on it; only one is needed to connect.
   If you check :guilabel:`Save connection settings`, these details will be remembered and filled for you automatically next time you open the dialog.
```

Update explanation to reflect several UI changes and new features in Spyder 4. Also, better reflect what's happening in the GIF (as I used the pre-filled details to avoid the GIF getting too long with typing them all in, while allowing more time for the user to see what the completed result should look like).

Reply:
I agree with this change.